cms/jinja2/component_overrides/header/_macro.njk

```diff
@@ -151,16 +151,11 @@
                         <div class="ons-container">
                             <ul class="ons-grid ons-grid-flex-gap ons-grid-flex-gap--40  navigation__key-links ons-list ons-list--bare">
                                 {% for item in params.navLinks.keyLinksList %}
-                                    {# In the new header, nav items can have text and no link - later we would want to test that both are present before outputting an li #}
-                                    {% if item.text %}
+                                    {% if item.text and item.url %}
                                         <li class="ons-grid__col ons-col-4@m ons-u-mb-no">
-                                            {% if item.url %}
-                                                <a href="{{ item.url }}" class="navigation__link">
-                                            {% endif %}
-                                                    <h2 class="ons-u-fs-s--b ons-u-mb-no navigation__heading avigation__heading--key-link">{{ item.text }}</h2>
-                                            {% if item.url %}
-                                                </a>
-                                            {% endif %}
+                                            <a href="{{ item.url }}" class="navigation__link">
+                                                <h2 class="ons-u-fs-s--b ons-u-mb-no navigation__heading avigation__heading--key-link">{{ item.text }}</h2>
```

Comment:
question: should the nesting be inverted? that is: `<h2><a></a></h2>` rather than `<a><h2>></h2></a>`?

Reply:
Good spot. In terms of HTML validation, it doesn't matter. I can't see an example of a header and link together in the current base template in the design system, but both gov.uk and w3c use links inside headings, so I'll switch it.

Reply:
W3C ref: https://www.w3.org/TR/WCAG20-TECHS/H80.html

Reply:
done.